tests/model/test_ocrd_mets.py

```diff
@@ -104,17 +123,82 @@ def test_find_all_files(sbb_sample_01):
+    assert len(mets.find_all_files(pageId='physical:id:~PHYS_0001..PHYS_0002')) == 1, '1 file for page "~PHYS_0001..PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:PHYS_0001,physical:~PHYS_0002')) == 17, '17 files for page "PHYS_0001,~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:id:PHYS_0001,physical:id:~PHYS_0002')) == 17, '17 files for page "PHYS_0001,~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:PHYS_0001..PHYS_0005,physical:~PHYS_0002')) == 18, '18 files for page "PHYS_0001..PHYS_0005,~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:id:PHYS_0001..PHYS_0005,physical:id:~PHYS_0002')) == 18, '18 files for page "PHYS_0001..PHYS_0005,~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:~PHYS_0002')) == 18, '18 files for page "~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical:id:~PHYS_0002')) == 18, '18 files for page "~PHYS_0002"'
+    assert len(mets.find_all_files(pageId='physical://PHYS_000(1|2)')) == 34, '34 files in PHYS_0001 and PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical:id://PHYS_000(1|2)')) == 34, '34 files in PHYS_0001 and PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical:~//PHYS_000(1|2)')) == 1, '1 file in PHYS_0005'
+    assert len(mets.find_all_files(pageId='physical:id:~//PHYS_000(1|2)')) == 1, '1 file in PHYS_0005'
+    assert len(mets.find_all_files(pageId='physical://PHYS_0001,physical://PHYS_0005')) == 18, '18 files in PHYS_001 and PHYS_0005 (two regexes)'
+    assert len(mets.find_all_files(pageId='physical:id://PHYS_0001,physical:id://PHYS_0005')) == 18, '18 files in PHYS_001 and PHYS_0005 (two regexes)'
+    assert len(mets.find_all_files(pageId='physical://PHYS_0005,physical:PHYS_0001..PHYS_0002')) == 35, '35 files in //PHYS_0005,PHYS_0001..PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical:id://PHYS_0005,physical:id:PHYS_0001..PHYS_0002')) == 35, '35 files in //PHYS_0005,PHYS_0001..PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical://PHYS_0005,physical:PHYS_0001..PHYS_0002')) == 35, '35 files in //PHYS_0005,PHYS_0001..PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical:id://PHYS_0005,physical:id:PHYS_0001..PHYS_0002')) == 35, '35 files in //PHYS_0005,PHYS_0001..PHYS_0002'
+    assert len(mets.find_all_files(pageId='physical:1..10')) == 35, '35 files in @ORDER range 1..10'
+    assert len(mets.find_all_files(pageId='physical:order:1..10')) == 35, '35 files in @ORDER range 1..10'
+    assert len(mets.find_all_files(pageId='physical:1..5')) == 35, '35 files in @ORDER range 1..10'
+    assert len(mets.find_all_files(pageId='physical:order:1..5')) == 35, '35 files in @ORDER range 1..10'
+    assert len(mets.find_all_files(pageId='physical:page 1..page 2,physical:5')) == 35, '35 in PHYS_0001,PHYS_0002,PHYS_0005'
+    assert len(mets.find_all_files(pageId='physical:orderlabel:page 1..page 2,physical:order:5')) == 35, '35 in PHYS_0001,PHYS_0002,PHYS_0005'
```

Comment:
Maybe we should put these tests on the website, since the syntax is getting powerful and complex and few users are probably aware of the extensive unit tests.

Reply:
Yes, this should be documented extensively. I recommend the Readme in src/ocrd_models or src/ocrd as nearest place. Next off, point to that from the ocrd-website pages (guides or Wiki).